includes/class-woocommerce-civicrm-manager.php

```diff
@@ -677,7 +676,7 @@ public function add_contribution( $cid, &$order ) {
 		 * @since 2.0
 		 * @param array $params The params to be passsed to the API
 		 */
-			$contribution = civicrm_api3( 'Contribution', 'create', apply_filters( 'woocommerce_civicrm_contribution_create_params', $params ) );
+			$contribution = civicrm_api3( 'Order', 'create', apply_filters( 'woocommerce_civicrm_contribution_create_params', $params, $order ) );
```

Comment:
Awesome 👍

Reply:
I have put the new code in and it seems to fix the problem of the financial contribution link, but the plugin also seems to be trying to create a membership which is failing as follows:
Error Details
=============
An error of type E_ERROR was caused in line 133 of the file /home/rjakn8hl475q/public_html/shop/wp-content/plugins/civicrm/civicrm/api/api.php. Error message: Uncaught CiviCRM_API3_Exception: [constraint violation: DB Error: constraint violation thrown

Maybe it is because I have set the WooCommerce product to set to Contribution Type = Membership in Civi although it has worked in the past.  Can you tell me what the code is trying to do with creating a membership record in Civi because I can't see anywhere in WooCommerce where this is set or how it is triggered?

Reply:
@ralphyke yes, setting the Financial Type to Member Dues seems to (obscurely) attempt to create a Membership, I think I will strip that out for now as it's not the best implementation.

Reply:
Ok, if you can strip that out and then I can put in the new version again, that would be great